### PR TITLE
imatix_gsl: init at 4.1

### PR DESCRIPTION
**Pull request by moosingin3space**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/imatix_gsl/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchgit, pcre } :
+
+stdenv.mkDerivation rec {
+    name = "imatix_gsl";
+    version = "4.1";
+    src = fetchgit {
```
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: Please use `fetchFromGitHub`, if possible

File: pkgs/development/tools/imatix_gsl/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        url = "https://github.com/imatix/gsl.git";
+	rev = "72192d0d9de17de08d9379602d6482b4e5d402d0";
+	sha256 = "0dyginj8l8f5rgbhlp4f3lhfz2ixdli9192l1bmc5jviklna4phg";
+    };
+
+    buildInputs = [ pcre ];
+
+    buildPhase = ''
```
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: Is it possible to do `cd src` in `preBuild` insead of overriding the entire `buildPhase` here?

Note that overriding a phase means you also miss out on potentially useful hooks/magic.

File: pkgs/development/tools/imatix_gsl/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    	cd src
+	make
+    '';
+
+    installPhase = ''
+    	make install DESTDIR=$out
```
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: Are you sure this is necessary? Note that you can pass `DESTDIR` via the `installFlags` attribute.

File: pkgs/development/tools/imatix_gsl/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    '';
+
+    meta.description = ''
```
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: Please provide at least `license`, `homepage` in addition to a short description. Especially `license` is required.

Also, consider adding yourself as the maintainer for this package. Adding orphaned packages to the collection doesn't really make any sense.

File: pkgs/development/tools/imatix_gsl/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    	Code construction tool developed by iMatix
```
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: Bikeshedding: the "developed by iMatix" part is redundant, as a potential user it does not help me decide whether I need this.

File: pkgs/development/tools/imatix_gsl/default.nix (hunk 1 above)
Comment (moosingin3space) [inline pkgs/development/tools/imatix_gsl/default.nix]: Should I simply amend the commit when pushing the fix?

File: pkgs/development/tools/imatix_gsl/default.nix (hunk 4 above)
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: Oh, I forgot: add `platforms` as well.

File: pkgs/development/tools/imatix_gsl/default.nix (hunk 1 above)
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: Yes, please. Or you could push each fixup as you go and squash before merge. Up to you.

File: pkgs/development/tools/imatix_gsl/default.nix (hunk 5 above)
Comment (moosingin3space) [inline pkgs/development/tools/imatix_gsl/default.nix]: Unfortunately, they don't provide much clarification for it. I'll just erase that.

File: pkgs/development/tools/imatix_gsl/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    preBuild = ''
+    	cd src
+    '';
+
+    installFlags = "DESTDIR=$(out)";
+
+    meta = with stdenv.lib; {
+    	 license = licenses.gpl3Plus;
+	 homepage = "https://github.com/imatix/gsl";
+	 description = ''
```
Comment (joachifm) [inline pkgs/development/tools/imatix_gsl/default.nix]: What about re-using upstream's description: "A universal code generator" (not that I really understand what that's supposed to be, but whatever).

**Comment by joachifm** [comment]
Looks good to me now. Please ping me when you're ready to merge (subscribers are not notified when the PR is updated).

**Comment by moosingin3space** [comment]
Ready to merge.

**Comment by joachifm** [comment]
Thank you

**Comment by moosingin3space** [comment]
This doesn't build on Hydra, and I have discovered why. How should I submit the fix?

**Comment by joachifm** [comment]
@moosingin3space it has been fixed at aaae243b2bbb6d1bf2d0b8f4de0df57f1055ded5
In general, fixes are submitted like any other PR :)

**Comment by moosingin3space** [comment]
Excellent. Thank you @joachifm for the fix!

**Pull request merged**